File: backend/chat-service/src/main/java/it/roadies/chat_service/data/dto/response/MessageResponseDTO.java
package it.roadies.chat_service.data.dto.response;

import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MessageResponseDTO {

    private UUID id;
    private UUID conversationId;
    private String senderId;
    private String senderName;
    private String content;
    private LocalDateTime timestamp;
    private boolean read;
}

File: android-app/app/src/main/java/it/roadies/android_app/client/models/chat/MessageResponseDTO.kt
package it.roadies.android_app.client.models.chat

import com.google.gson.annotations.SerializedName

data class MessageResponseDTO(
    @SerializedName("id")
    val id: String,

    @SerializedName("conversationId")
    val conversationId: String,

    @SerializedName("senderId")
    val senderId: String,

    @SerializedName("content")
    val content: String,

    @SerializedName("timestamp")
    val timestamp: String,

    @SerializedName("isRead")
    val isRead: Boolean
)